services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusControllerITest.kt

```diff
@@ -462,6 +462,32 @@ class HakemusControllerITest(@Autowired override val mockMvc: MockMvc) : Control
                 hakemusService.hankkeenHakemukset(HANKE_TUNNUS)
             }
         }
+
+        @Test
+        fun `With areas included return application areas`() {
+            val cableReportApplicationResponses =
+                HakemusFactory.createSeveral(2, applicationType = ApplicationType.CABLE_REPORT)
+            val excavationNotificationResponses =
+                HakemusFactory.createSeveral(
+                    2, applicationType = ApplicationType.EXCAVATION_NOTIFICATION)
+            val hakemukset = cableReportApplicationResponses + excavationNotificationResponses
+            every { hakemusService.hankkeenHakemukset(HANKE_TUNNUS) } returns hakemukset
+            every {
+                authorizer.authorizeHankeTunnus(HANKE_TUNNUS, PermissionCode.VIEW.name)
+            } returns true
+
+            val response: HankkeenHakemuksetResponse =
+                get("$url?areas=true").andExpect(status().isOk).andReturnBody()
+
+            assertThat(response.applications).isNotEmpty()
+            val expected =
+                HankkeenHakemuksetResponse(hakemukset.map { HankkeenHakemusResponse(it, true) })
+            assertThat(response).isEqualTo(expected)
```

Comment:
This doesn't actually test if there are any areas in the response. The `HankkeenHakemusResponse` can ignore the parameter completely and always set the areas to null and the test will still pass.

Maybe check that the response has the same areas as `hakemukset`?

```suggestion
            for (i in 0..3) {
                assertThat(response.applications[i].applicationData.areas)
                    .isNotNull()
                    .single()
                    .isEqualTo(hakemukset[i].applicationData.areas?.single())
            }
```

Reply:
Thanks. Fixed this.